### PR TITLE
Aligned label and head predictions in predictWithScores. This is needed

### PR DESCRIPTION
**Pull request by MihaiSurdeanu**
for ensemble parsing.

### CODE REVIEW

File: encoder/src/main/scala/org/clulab/scala_transformers/encoder/LinearLayer.scala [hunk 1]
```diff
@@ -206,7 +254,7 @@ class LinearLayer(
         val labelsAndScores = labels.zip(scores)
 
         // keep scores in descending order (largest first) 
-        labelsAndScores.sortBy(_._2)
+        labelsAndScores.sortBy(- _._2) // - score guarantees sorting in descending order of scores
```
Comment (kwalcock) [inline encoder/src/main/scala/org/clulab/scala_transformers/encoder/LinearLayer.scala:231]: Sorry about losing the -.  I had it right once and then the -_ must have looked like a typo.

File: encoder/src/main/scala/org/clulab/scala_transformers/encoder/LinearLayer.scala [hunk 2]
```diff
@@ -134,41 +166,57 @@ class LinearLayer(
...
+                            batchHeads: Option[Array[Array[Array[Int]]]] = None,
                             batchMasks: Option[Array[Array[Boolean]]] = None): Array[Array[Array[(String, Float)]]] = {
     assert(batchHeads.isDefined)
     assert(batchMasks.isDefined)
     val indexToLabel = labelsOpt.getOrElse(throw new RuntimeException("ERROR: can't predict without labels!"))
 
+    // dimensions: sent in batch x token in sentence x label per candidate head
     val outputBatch = new Array[Array[Array[(String, Float)]]](inputBatch.length)
 
+    // TODO: maybe the triple for loop below can be improved?
```
Comment (kwalcock) [inline encoder/src/main/scala/org/clulab/scala_transformers/encoder/LinearLayer.scala]: I will TODO this.

**Review by kwalcock — APPROVED** [review]
But see also #40 that might be merged here before this is in turn merged.

**Comment by MihaiSurdeanu** [comment]
Both can be merged! Thanks @kwalcock !

**Pull request merged**